pkg/controller/standalone/standalone.go

```diff
@@ -44,9 +45,32 @@ func Server(opts ServerOptions) error {
 	// for validation and conversion purposes.
 	swarmResourceBackend := interfaces.NewSwarmAPIClientShim(dclient)
 
+	// We can access the swarm Resources backend even without changing docker
+	// engine code,
+	//
+	// NOTE(dperny): this is a hack, and there are lots of caveats
+	// * The swarm control socket may not always be located in the assumed
+	//   location
+	// * The client connection we use for the control client may somehow
+	//   timeout or otherwise error, and there is currently no recovery from
+	//   that case.
+	// * The swarm control socket is generally not considered to be part of the
+	//   public API
+	// * The swarm client connection cannot generate events; those must still
+	//   be provided by the stacks shim.
+	// * This will only work on an engine with the latest Swarmkit code
+	//   vendoring. Otherwise, the resources API is not yet available.
+	//
+	// Now, with all that in mind, create a swarmkit control client...
+	cc, err := newControlClient()
+	if err != nil {
+		return err
+	}
+	stackStore := store.New(cc)
+
 	// Create the underlying storage for stacks and swarmstacks as an
 	// in-memory store.
-	stackStore := interfaces.NewFakeStackStore()
+	// stackStore := interfaces.NewFakeStackStore()
```

Comment:
Just wondering if it would make sense to have this be driven by presence of the underlying API?  (or API version maybe?)

I agree with @alexmavr that until the moby PR merges and we start getting builds of engine with it integrated, if we merge this PR it complicates local testing scenarios.

Reply:
🤔 yeah you're right.